### PR TITLE
Api refactor

### PR DESCRIPTION
**Pull request by shichong-97**
### Summary 

Major refactor + reorganization of the API endpoints to make our backend more pleasant to work with. Many APIs which had overlapping functionality were combined into a single API (with customizable parameters), and APIs that are no longer being used are deleted.

Also created documentation for most of the APIs, which can be found at cuevents2019.docs.apiary.io.

Will spend the work day today working with people from mobile and web to update the endpoints used and to make sure the updated APIs continue to provide the same functionalities as before.

### CODE REVIEW

File: events_frontend/src/Main.js [hunk 1]
```diff
@@ -132,21 +132,21 @@ class Main extends Component {
...
+                  path={obj.route}
+                  component={obj.component}
+                />
+              ))}
+              {Object.values(routes.auth).map(obj => (
+                <Redirect key={obj.route} from={obj.route} to={"/login/"} />
+              ))}
+              <Redirect to={"/"} />
+            </Switch>
+          )}
```
Comment (SamChou19815) [inline events_frontend/src/Main.js]: You should avoid this indentation only change. It is very likely to cause merge conflicts, and your change is not consistent with the rest of the code style

File: events_frontend/src/Profile.js [hunk 2]
```diff
@@ -104,7 +104,14 @@ class Profile extends Component {
...
     this.setState({ profileUpdated: false });
-    let orgData = this.state,
+    let { bio, email, name, tags, website } = this.state,
+      orgData = {
+        bio,
+        email,
+        name,
+        tags,
+        website
+      },
```
Comment (SamChou19815) [inline events_frontend/src/Profile.js]: Indentation here is not consistent with the rest

File: package.json [hunk 3]
```diff
@@ -75,10 +75,8 @@
     "backend-windows": "python3 \"events_backend\\manage.py\" runserver",
     "heroku-postbuild": "webpack --config events_frontend/webpack.prod.js && python3 events_backend/manage.py collectstatic --noinput",
     "heroku-postbuild-windows": "webpack --config \"events_frontend\\webpack.prod.js\" && python3 \"events_backend\\manage.py\" collectstatic --noinput",
-    "python-migrate": "python events_backend/manage.py makemigrations app && python events_backend/manage.py migrate",
-    "python3-migrate": "python3 events_backend/manage.py makemigrations app && python3 events_backend/manage.py migrate",
-    "python-migrate-windows": "python \"events_backend\\manage.py\" makemigrations app && python \"events_backend\\manage.py\" migrate",
-    "python3-migrate-windows": "python3 \"events_backend\\manage.py\" makemigrations app && python3 \"events_backend\\manage.py\" migrate"
+    "python-migrate": "python3 events_backend/manage.py makemigrations app && python events_backend/manage.py migrate",
+    "python-migrate-windows": "python3 \"events_backend\\manage.py\" makemigrations app && python \"events_backend\\manage.py\" migrate"
```
Comment (SamChou19815) [inline package.json]: Reason for the change?

Comment (shichong-97) [reply package.json]: just for standardization, since we enforced the usage of "python3" and took "3" out of all other script names

**Pull request merged**